game-core/src/main/java/games/strategy/triplea/ui/DicePanel.java

```diff
@@ -44,15 +46,22 @@ public void setDiceRoll(final DiceRoll diceRoll) {
     SwingAction.invokeNowOrLater(() -> {
       removeAll();
       for (int i = 1; i <= data.getDiceSides(); i++) {
-        final List<Die> dice = diceRoll.getRolls(i);
+        final int index = i;
+        final List<Die> dice = Optional.ofNullable(diceRoll)
+            .map(roll -> roll.getRolls(index))
+            .orElse(Collections.emptyList());
```

Comment:
OrElseGet should be preferred here, even if the difference is minimal.

Reply:
That is pretty subjective. `Collections.emptyList()` evaluates to a constant reference vs creating a new supplier instance each time. It's almost the same difference as using `Collections.emptyList()` vs `new AraryList<>()`, one creates a new object each time.


It is also hard to say which is more impactful on clarity. `orElseGet` has more syntax and could lead one to wonder why the extra syntax was used.